src/library_wasi.js

```diff
@@ -519,8 +519,10 @@ var WasiLibrary = {
                  FS.isLink(stream.mode) ? {{{ cDefs.__WASI_FILETYPE_SYMBOLIC_LINK }}} :
                  {{{ cDefs.__WASI_FILETYPE_REGULAR_FILE }}};
 #else
-      // hack to support printf in SYSCALLS_REQUIRE_FILESYSTEM=0
-      var type = fd == 0 || fd == 1 || fd == 2 ? {{{ cDefs.__WASI_FILETYPE_CHARACTER_DEVICE }}} : abort();
+      // Hack to support printf in SYSCALLS_REQUIRE_FILESYSTEM=0. We support at
+      // least stdin, stdout, stderr in a simple way.
+      assert(fd == 0 || fd == 1 || fd == 2);
```

Comment:
I think this line needs to be wrapped in `#if ASSERTIONS` .. at least for `MINIMAL_RUNTIME`.  Se e the bug I just opened about this: https://github.com/emscripten-core/emscripten/issues/19453

Reply:
Oh, good catch, fixing.